### PR TITLE
[CGAL] Upgrade to 5.3

### PR DESCRIPTION
**Pull request by maxGimeno**
Upgrade CGAL to version 5.3

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@maxGimeno ,Thanks for your contribution.
 upgrade to the 5.3-beta1 pre-release version, our policy is not recommended, because the pre-release version is unstable and there may be many problems

**Comment by maxGimeno** [comment]
@JonLiu1993 don't worry, it will stay a draft until the release of the CGAL 5.3 version, then I'll update the PR to the stable version and un-draft it. For now it was just to be sure that in the current state we can pass your CI.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]
Please add 'FEATURES' in function vcpkg_check_features like:
````
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
    qt WITH_CGAL_Qt5
)
````

**Comment by JonLiu1993** [comment]
@maxGimeno ,After completing the modification, please run the command "./vcpkg x-add-version cgal --overwrite-version", and then submit the changes again.

**Review by JonLiu1993 — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR!

**Review by PhoebeHui — APPROVED** [review]

**Review by vicroms — APPROVED** [review]
Please resolve merge conglicts, otherwise LGTM.

File: versions/c-/cgal.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "02c3a0b581425d69dc21787d87e14c53a07e33ee",
-      "version": "5.2.2",
-      "port-version": 1
```
Comment (JonLiu1993) [inline versions/c-/cgal.json:6]: @maxGimeno, we cannot delete this, we should add a new "version" for the new pr

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**